specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json

```diff
@@ -3332,6 +3332,475 @@
+            }
+          },
+          "304": {
+            "description": "Not modified."
+          },
+          "default": {
+            "description": "An error response received from the Azure Data Factory service.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "dataFlows"
+        ],
+        "operationId": "DataFlows_Delete",
+        "x-ms-examples": {
+          "DataFlows_Delete": {
+            "$ref": "./examples/DataFlows_Delete.json"
+          }
+        },
+        "description": "Deletes a data flow.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/factoryName"
+          },
+          {
+            "$ref": "#/parameters/dataFlowName"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK."
+          },
+          "204": {
+            "description": "No Content."
+          },
+          "default": {
+            "description": "An error response received from the Azure Data Factory service.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataFactory/factories/{factoryName}/dataflows": {
+      "get": {
+        "tags": [
+          "dataFlows"
+        ],
+        "operationId": "DataFlows_ListByFactory",
+        "x-ms-examples": {
+          "DataFlows_ListByFactory": {
+            "$ref": "./examples/DataFlows_ListByFactory.json"
+          }
+        },
+        "description": "Lists data flows.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/factoryName"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK.",
+            "schema": {
+              "$ref": "#/definitions/DataFlowListResponse"
+            }
+          },
+          "default": {
+            "description": "An error response received from the Azure Data Factory service.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataFactory/factories/{factoryName}/createDataFlowDebugSession": {
```

Comment:
>createDataFlowDebugSession [](start = 128, length = 26)

You have several POST actions that perform various lifecycle operations (create, delete, query) on data flow debug sessions. Heavily relying on POST is generally an anti-pattern in ARM.

Have you considered modeling dataFlowDebugSessions as a proxy resource? If yes, why did you choose the POST approach?

Reply:
We don't want to treat dataFlowDebugSession as a proxy resource since it's a just a temporary session which users can perform some debugging actions. "Create", "AddDataFlow" and "Execute" needs to be POST actions though I can modify query to be get if ARM prefers that way.

Reply:
It seems that the sessionId is assigned by the backend, so it'll have to be POST.

---
In reply to: [325409815](https://github.com/Azure/azure-rest-api-specs/pull/7088#discussion_r325409815) [](ancestors = 325409815)

Reply:
yep, reverted to post